angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html

```diff
@@ -29,25 +29,28 @@ <h4 id="appTitle_portlet.title-{{::portlet.nodeId}}" aria-labelledby="appTitle_p
     </div>
     <a class="btn btn-default launch-app-button" href="{{::portlet.url}}" target="{{::portlet.target}}">Launch full app</a>
   </div>
-  
+
   <div ng-if="'OPTION_LINK' === widgetCtrl.portletType(portlet)">
     <option-link app="portlet" config="portlet.widgetConfig"></option-link>
   </div>
-  
+
   <div ng-if="'WEATHER' === widgetCtrl.portletType(portlet)">
     <weather app="portlet" config="portlet.widgetConfig"></weather>
   </div>
-  
+
   <div ng-if="'RSS' === widgetCtrl.portletType(portlet)">
     <rss app="portlet" config="portlet.widgetConfig"></rss>
   </div>
-  
+
   <div ng-if="'GENERIC' === widgetCtrl.portletType(portlet)">
     <div ng-controller="GenericWidgetController as genericWidgetCtrl">
+        <div ng-if="loading" id="loading">
+            <loading-gif data-object='content' data-empty='isEmpty'></loading-gif>
+        </div>
```

Comment:
The original idea of the "loading state" was that the widget would have the option to have a loading state. I do like this uniformed approach tho. Makes way more sense.